airbyte-integrations/connectors/source-google-analytics-data-api/source_google_analytics_data_api/source.py

```diff
@@ -193,9 +201,24 @@ def parse_response(
         metrics_type_map = {h.get("name"): h.get("type") for h in r.get("metricHeaders", [{}])}
 
         for row in r.get("rows", []):
-            yield self.add_primary_key() | self.add_property_id(self.config["property_id"]) | self.add_dimensions(
-                dimensions, row
-            ) | self.add_metrics(metrics, metrics_type_map, row)
+            record = {
+                "property_id": self.config["property_id"],
+                **self.add_dimensions(dimensions, row),
+                **self.add_metrics(metrics, metrics_type_map, row),
+            }
+
+            # https://github.com/airbytehq/airbyte/pull/26283
+            # We pass the uuid field for synchronizations which still have the old
+            # configured_catalog with the old primary key. We need it to avoid of removal of rows
+            # in the deduplication process. As soon as the customer press "refresh source schema"
+            # this part is no longer needed.
```

Comment:
how will we know when we can remove this code block?

Reply:
I guess we need help from TSC team to communicate with users and set due date, and after this date we can remove this code block.